pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -280,16 +300,22 @@ func (c *GuardController) sync(ctx context.Context, syncCtx factory.SyncContext)
 	return utilerrors.NewAggregate(errs)
 }
 
-func IsSNOCheckFnc(infraLister configv1listers.InfrastructureLister) func() (bool, error) {
-	return func() (bool, error) {
-		infraData, err := infraLister.Get("cluster")
+// IsSNOCheckFnc creates a function that checks if the topology is SNO
+func IsSNOCheckFnc(infraInformer configv1informers.InfrastructureInformer) func() (bool, bool, error) {
+	// is, unknown, error
+	return func() (bool, bool, error) {
+		if !infraInformer.Informer().HasSynced() {
+			// Do not return transient error
+			return false, true, nil
+		}
+		infraData, err := infraInformer.Lister().Get("cluster")
 		if err != nil {
-			return false, fmt.Errorf("Unable to list infrastructures.config.openshift.io/cluster object, unable to determine topology mode")
+			return false, true, fmt.Errorf("Unable to list infrastructures.config.openshift.io/cluster object, unable to determine topology mode")
```

Comment:
shouldn't we return `false` here ?

Reply:
When the `err` is not nil, the second returned value has no use so it can have arbitrary value. Given the isSNO condition is known only when the infrastructure object has its status and the topology properly set, returning `true` is more suitable.